src/bootstrap/compile.rs

```diff
@@ -101,6 +101,10 @@ impl Step for Std {
             || builder.config.keep_stage_std.contains(&compiler.stage)
         {
             builder.info("Warning: Using a potentially old libstd. This may not behave well.");
+
+            copy_third_party_objects(builder, &compiler, target);
+            copy_self_contained_objects(builder, &compiler, target);
```

Comment:
this is fine to make sure the try build it works, but once you've finished testing it would be nice to move this to be part of `StdLink`

Reply:
ah never mind, they need to be part of `target_deps` to avoid changing the stamp file. that's unfortunate :/ but no changes needed, sorry for the noise

Reply:
Yeah I did that locally, but then I noticed that the output of the copied objects is needed to set the target deps, which is done before StdLink.